### PR TITLE
Changing  a few more instances of api.Pod to *api.Pod

### PR DESCRIPTION
**Pull request by krousey**
Cleaning up a few remaining instances of api.Pod shallow copies. Again, I drew the line at modifying api.PodList.Items. #554

### CODE REVIEW

**Comment by krousey** [comment]
@lavalamp Since you reviewed the previous one too.

File: pkg/scheduler/listers.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type PodLister interface {
 // FakePodLister implements PodLister on an []api.Pods for test purposes.
 type FakePodLister []*api.Pod
 
-// List returns []api.Pod matching a query.
+// List returns []*api.Pod matching a query.
```
Comment (lavalamp) [inline pkg/scheduler/listers.go]: Oops, this is probably due to me rebasing poorly. Sorry!

Comment (krousey) [inline pkg/scheduler/listers.go]: No. I missed it the first time. I'm fairly certain of it.

File: pkg/kubectl/describe.go [hunk 2]
```diff
@@ -475,7 +475,7 @@ func (d *NodeDescriber) Describe(namespace, name string) (string, error) {
 		if pod.Spec.Host != name {
 			continue
 		}
-		pods = append(pods, pod)
+		pods = append(pods, &pod)
```
Comment (lavalamp) [inline pkg/kubectl/describe.go]: This is a bug-- every pod will have the same address.

Comment (krousey) [inline pkg/kubectl/describe.go]: After we just talked about this too. I was careful not to do it in the other pieces of code that dealt with PodList.Items. Oh well. Thanks for catching.

**Comment by lavalamp** [comment]
LGTM - can you squash your commits?

**Comment by krousey** [comment]
Squashed.

**Comment by lavalamp** [comment]
Thanks!

**Pull request merged**